src/Zoomba/APILibrary.py

```diff
@@ -225,14 +231,18 @@ def validate_response_contains_correct_number_of_items(self, json_actual_respons
             number_of_items = number_of_items.upper()
             if number_of_items == "IGNORE":
                 return True
-            else:
-                assert False, "Passed a unicode or string value, function expects a number or string 'IGNORE'."
+        elif type(number_of_items) is int:
+            pass
+        else:
+            zoomba.fail("Did not pass number or string value, function expects a number or string 'IGNORE'.")
+            return
 
         if type(actual_response_dict) is list:
-            assert len(actual_response_dict) == number_of_items, 'API is returning ' + str(
-                len(actual_response_dict)) + ' instead of the expected ' + str(number_of_items) + ' result(s).'
+            if len(actual_response_dict) != number_of_items:
+                zoomba.fail('API is returning ' + str(
+                    len(actual_response_dict)) + ' instead of the expected ' + str(number_of_items) + ' result(s).')
         else:
-            raise TypeError("The response is not a list:\nActual Response:\n" + str(actual_response_dict))
```

Comment:
raising a TypeError is different than an assert.  Are we sure that the zoomba.fail won't break any of the code depending on a typeerror being thrown?

Reply:
The main end all difference is that instead of returning 
"TypeError: The response is not a list:"
is that we will return
"The response is not a list:"
functionally speaking.

There is also a difference in that code exceptions are also displayed at the top of the log.html and in this case won't be displayed. Both pathways will fail the test.

So in the case that someone is deliberately sending in a non list parameter and expecting it to fail would affect their test. That isn't a normal test case however. So this is definitely an exception and could be changed, but I would like to push towards consistency in how we fail our tests.

Reply:
works for me.  can we make a comment in the Automation channel that this change is on purpose?